### PR TITLE
DSDEEPB-521 Fix race condition causing calls to be started multiple times.

### PR DESCRIPTION
**Pull request by mcovarr**

### CODE REVIEW

**Comment by scottfrazer** [comment]
Nice and terse, though unfortunately not terse enough to avoid the dreaded rebase.

File: src/main/scala/cromwell/engine/ExecutionStore.scala [hunk 1]
```diff
@@ -20,14 +20,19 @@ class ExecutionStore(namespace: WdlNamespace) {
...
-  def runnableCalls: Iterable[Call] = {
-    for {
+  def startRunnableCalls: Iterable[Call] = {
+    val runnableCalls = for {
       callEntry <- table if callEntry._2 == ExecutionStatus.NotStarted
       call = callEntry._1 if callIsRunnable(call)
     } yield call
+
+    runnableCalls foreach { call =>
+      table += call -> ExecutionStatus.Starting
```
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/ExecutionStore.scala]: This is adding a call in Starting state, right? Should it be replacing the NotStarted one?

Comment (mcovarr) [inline src/main/scala/cromwell/engine/ExecutionStore.scala]: `table` is a `Map[Call, ExecutionStatus]` which is known to already contain an entry for this particular `Call`, so this is a replacement.

Comment (mcovarr) [inline src/main/scala/cromwell/engine/ExecutionStore.scala]: I'll add a comment as that wasn't clear.

Comment (geoffjentry) [inline src/main/scala/cromwell/engine/ExecutionStore.scala]: Don't mind me, I was just being dim

**Pull request merged**